src/components/Sidebar.svelte

```diff
@@ -1,45 +1,38 @@
 <script lang="ts">
-    import { createEventDispatcher, onMount, onDestroy } from 'svelte';
+    import { createEventDispatcher} from 'svelte';
     export let sidebarOpen: boolean;
-    export let toggleButton;
+    export let toggleButton: HTMLElement | null;
     export let selectFranchise: (id: number) => void;
 
     const dispatch = createEventDispatcher();
-    let sidebar;
+    let sidebar: HTMLElement | null;
 
-    function handleSelectFranchise(id) {
+    function handleSelectFranchise(id: number) {
         selectFranchise(id);
+        dispatch('franchiseSelected', { id });
         sidebarOpen = !sidebarOpen;
     }
 
-    function handleClickOutside(event) {
-        if (!sidebar.contains(event.target) && !toggleButton.contains(event.target)) {
-            sidebarOpen = false;
-        }
-    }
-
-    function closeSidebar() {
+    function toggleSidebar() {
         sidebarOpen = !sidebarOpen;
     }
 
-    onMount(() => {
-        document.addEventListener('click', handleClickOutside);
-    });
-
-    onDestroy(() => {
-        document.removeEventListener('click', handleClickOutside);
-    });
+    function handleKeydown(event: KeyboardEvent) {
+        if (event.key === 'Escape' || event.key === 'Enter' || event.key === ' ') {
```

Comment:
Here the `event.code` can also be implemented

Reply:
Yes I've also changed this one :

https://github.com/BowenGeense/MovieShowcase/pull/3/files#diff-5fa8bc1e300230c2638d059cc9cd608ab21c79c6733942d5ddb7ca00bae85311